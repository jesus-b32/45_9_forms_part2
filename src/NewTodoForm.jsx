import React, {useState} from "react";
import { v4 as uuid } from 'uuid';


function NewTodoForm ({ addTodo }) {
        /**
     * formDate is an object that holds 
     *  task
     *  */
    const [formData, setFormData] = useState({task: ''});

    /** Update local state w/curr state of input elem */
    const handleChange = evt => {
        const { name, value } = evt.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }));
    };
    
        /** Send {backgroundColor, width, height} to parent
    *    & clear form. */
    const handleSubmit = evt => {
        evt.preventDefault();
        addTodo({ ...formData, id: uuid() });
        setFormData({ task: "" });
    };
    
    return (
        <div>
            <form onSubmit={handleSubmit}>
                <label htmlFor="task">Task: </label>
                <input
                onChange={handleChange}
                type="text"
                name="task"
                value={formData.task}
                id="task"
                />
                <button id="newTodoButton">Add a TODO!</button>
            </form>
        </div>
    );

}

export default NewTodoForm;