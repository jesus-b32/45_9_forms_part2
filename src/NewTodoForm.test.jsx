import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';
import { describe, it, expect, vi } from 'vitest';

describe('NewTodoForm', () => {
    //smoke test
    it('renders App component', () => {
        render(<NewTodoForm/>);
    })

    //snapshot test
    it('matches snapshot', () => {
        const {asFragment} = render(<NewTodoForm/>);
        expect(asFragment()).toMatchSnapshot();
    })

    it("runs the addTodo function on form submit", function() {
        const createMock = vi.fn();
        const { getByText } = render(<NewTodoForm addTodo={createMock} />);
        const createButton = getByText("Add a TODO!");
        fireEvent.click(createButton);
        expect(createMock).toHaveBeenCalled();
    });
})