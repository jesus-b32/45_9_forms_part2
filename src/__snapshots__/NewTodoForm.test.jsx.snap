// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewTodoForm > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="task"
      >
        Task: 
      </label>
      <input
        id="task"
        name="task"
        type="text"
        value=""
      />
      <button
        id="newTodoButton"
      >
        Add a TODO!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
